src/python/pants/backend/python/rules/pex.py

```diff
@@ -67,13 +75,79 @@ def create_from_adaptors(
         return PexInterpreterConstraints(constraint_set=tuple(sorted(interpreter_constraints)))
 
 
+@dataclass(frozen=True)
+class PexRequirementConstraints:
+    """A collection of paths to constraint files and/or Requirement-style strings."""
+
+    constraint_file_paths: Tuple[str, ...] = ()
+    raw_constraints: Tuple[str, ...] = ()
+    generated_file_name: ClassVar[str] = "requirement_constraints.generated.txt"
+
+    @classmethod
+    def create_from_raw_constraints(cls, constraints: Iterable[str]) -> "PexRequirementConstraints":
+        """The constraints may be paths to constraint .txt files and/or requirement strings."""
+        constraint_file_paths = set()
+        raw_constraints = set()
+        for constraint in constraints:
+            if PurePath(constraint).suffix == ".txt":
+                constraint_file_paths.add(constraint)
```

Comment:
I couldn't find in the [Pip spec for requirements files](https://pip.pypa.io/en/stable/reference/pip_install/#requirements-file-format) if it's possible for constraint and requirements files to not be text files. But, there is a very heavy convention around it so this seemed like a safe heuristic to use. 

We mention `.txt` files in all the documentation to be extra helpful.

Reply:
I have a feeling that I've seen constraint files that don't end in `.txt`, e.g., at Color. This seems like it might make a bad impression: "hey, you have to rename your constraints file for this to work". 

How about detecting `=`, `>` or `<` in the string, as an alternate heuristic?

Reply:
Originally, I thought we couldn’t simply check for `=`, `>`, or `<` because requirement strings can simply be the name of the distribution, eg `isort`. But I suppose that doesn’t make sense in the context of _constraints_ because that would not actually constrain anything. So, I think your heuristic would work and is an improvement.

Reply:
I have a feeling that I've seen constraint files that don't end in `.txt`. This seems like it might make a bad impression: "hey, you have to rename your constraints file for this to work". 

How about detecting `=`, `>` or `<` in the string, as an alternate heuristic?